lib/content-store/src/store/local.rs

```diff
@@ -0,0 +1,46 @@
+use crate::hash::ContentHash;
+use crate::store::{Store, StoreResult};
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+use std::collections::HashMap;
+
+/// A kind of content store that operates entirely in memory.
+#[derive(Default, Debug)]
+pub struct LocalStore(HashMap<ContentHash, Vec<u8>>);
+
+#[async_trait::async_trait]
+impl Store for LocalStore {
+    fn is_empty(&self) -> bool {
+        self.0.is_empty()
+    }
+
+    fn len(&self) -> usize {
+        self.0.len()
+    }
+
+    fn add<T>(&mut self, object: &T) -> StoreResult<ContentHash>
+    where
+        T: Serialize + ?Sized,
+    {
+        let value = serde_json::to_vec(object)?;
+        let key = ContentHash::new(&serde_json::to_vec(&value)?);
```

Comment:
I'm curious why the two layers of `serde_json::to_vec` here. Shouldn't the key be `ContentHash::new(&value)`, instead of `ContentHash::new(serde_json::to_vec(&value)?`), since `object` was already converted to a `Vec<u8>` of the serialized bytes?

Reply:
That... is an accident.

Reply:
It should be just one.